coderd/tailnet.go

```diff
@@ -37,6 +37,9 @@ func init() {
 	if !valid {
 		panic("dev error: default transport is the wrong type")
 	}
+	// We do not want to respect the proxy settings from the environment, since
+	// all network traffic happens over wireguard.
+	tailnetTransport.Proxy = nil
```

Comment:
Won't this disable the proxy for everything that uses `http.DefaultTransport`?  Probably need to clone it before making this change!

Reply:
Yeah, that's a great point. Good catch